test/test.js

```diff
@@ -53,6 +53,24 @@ test('generate screenshots', async t => {
 	t.true((await getStream.buffer(streams[0])).length > 1000);
 });
 
+test('save filename with hash', async t => {
+	const streams = await new Pageres()
+		.src('example.com/#/', ['480x320'])
+		.src('example.com/#/@user', ['480x320'])
+		.src('example.com/#/product/listing', ['480x320'])
+		.src('example.com/#!/bang', ['480x320'])
+		.src('example.com#readme', ['480x320'])
+		.run();
+
+	t.is(streams.length, 5);
+	t.is(streams[0].filename, 'example.com#-480x320.png');
```

Comment:
hashes that exactly match `#`, `#/` and `#!/` will become `example.com-480x320.png` now.